docs/source/notes/cuda.rst

```diff
@@ -42,6 +42,16 @@ Below you can find a small example showcasing this::
         d = torch.randn(2).cuda(2)
         # d.get_device() == 2
 
+Memory management
+-----------------
+
+PyTorch use a caching memory allocator to speed up memory allocations. This
+allows very fast memory allocation for small sizes. However, the unused memory
```

Comment:
I'd remove the sentence about small sizes. Unlike CPU allocators, the speed-up of caching CUDA allocations isn't particularly about the size of the allocation. It's about avoiding synchronizations on free.